### DISCUSSION TITLE
Could you provide your firefly docker images also with debian buster as base

### DISCUSSION THREAD

**Discussion by rerobade** [firefly-iii/firefly-iii#7205] [General]
Hi @JC5 ,

I really like firefly and I'm currently running the latest pre PHP 8.2 versions on my Synology NAS. Today I tried to update firefly and the data-importer to version 6 & 1 but I encountered the following error: 
`Function not implemented: AH00141: Could not initialize random number generator`
After all migrations the container automatically restarted and looped with starting -> migrating -> complainig -> starting -> ... :
Reason for this is not firefly but the "debian bullseye" base image you used for your docker images which is causing errors due to the old Linux Kernel Version used in Synology DSM 7.1. I've seen this issue also already in the past when I played with nextcloud on docker and the solution was to build a new Image based on debian buster which is on one hand much older then bullseye but compatible with the old Linux Kernel on DSM 7.1. For this reason I would like to ask you if you could provide also firefly 6 images based on debian buster, maybe as "version-6.0.2-buster" tag. 
I'm sure every Firefly-Fan with Synology Hardware would be happy about it. I would appreciate it.

Thanks & Regards René

**Comment by JC5** [comment]
Hi René,

No. But the the Docker images are also open source, and you are free to adjust the scripts and build Firefly III yourself.

Check out:

- [The base image](https://dev.azure.com/Firefly-III/_git/BaseImage)
- [The Firefly III image](https://dev.azure.com/Firefly-III/_git/MainImage)

**Reply by rerobade** [reply]
Okay, then I will do it myself :-) can you please also share the link for the data-importer repo? Or is this integrated in the MainImage Repo?

Regards René

**Reply by JC5** [reply]
https://dev.azure.com/Firefly-III/_git/ImportToolImage

**Reply by rerobade** [reply]
Hi @JC5 ,

I just wanted to give you the feedback that it worked with buster as base image.

Regards René

**Reply by JC5** [reply]
If you have anything you can share for other users, that would be great!

**Reply by rerobade** [reply]
Yes I only did some small adjustments to your original repos and build everything with drone ci on my local machine. I just created copies of your repos to show my changes in single commits:

### BaseImage 

- adjusted generate-docker-files.php to make the Image Base independent from PHP Version
- adjusted the apache-8.2-config to avoid docker context trouble when trying to add files from parent directories
- created a .drone.yml to build everything and push the BaseImage to my local registry
- Commit: https://github.com/rerobade/Firefly-III-BaseImage-Synology/commit/8936d1de1a48b3fb86fa54c5ba646b28253a93e5

### MainImage & ImportToolImage

- adjusted Dockerfile to accept only one parameter the base image and to make it compatible with Docker >= 17.05 (I assume that your pipeline based on ubuntu 16.04 is using a much older docker version. This could also be a security risk to be honest)
- created a .drone.yml to build everything and push the images to my local registry
- Commit: https://github.com/rerobade/Firefly-III-MainImage-Synology/commit/54fe555741013a07d456a26c02a2529fdb5b01e4
- Commit: https://github.com/rerobade/Firefly-III-ImportToolImage-Synology/commit/3cc377081a5ab03efdefaacfe15403776e60ddd2

As I said I just copied your repos to share my changes. If you wish that I make further adjustments or delete sth. additional to let it look less official please reach out to me.

Regards René

**Reply by rerobade** [reply]
Hi @Bolela76, my image is only available in my private registry. With the example commits I provided you can also build the images on your environment and use it.

**Comment by Zap0** [comment]
Oh, if firefly doesn't work I'll stay on DSM 6 then.
